bodhi/server/consumers/masher.py

```diff
@@ -979,3 +981,41 @@ def run(self):
         else:
             self.success = True
         return out, err, returncode
+
+NUM_TO_KEEP = 10
+
+
+def clean_up():
+    """
+    Delete any repo mashes that are older than the newest 10 from each repo series.
+    """
+    mash_dir = config.config['mash_dir']
+
+    # This data structure will map the beginning of a group of dirs for the same repo to a list of
+    # the dirs that start off the same way.
+    pattern_matched_dirs = collections.defaultdict(list)
+
+    for directory in [d for d in os.listdir(mash_dir) if os.path.isdir(os.path.join(mash_dir, d))]:
+        # If this directory ends with a float, it is a candidate for potential deletion
+        try:
+            split_dir = directory.split('-')
+            float(split_dir[-1])
+        except ValueError:
+            # This directory didn't end in a float, so let's just move on to the next one.
+            continue
+
+        pattern = directory.replace(split_dir[-1], '')
+        pattern_matched_dirs[pattern].append(directory)
+
+    dirs_to_delete = []
+
+    for dirs in pattern_matched_dirs.values():
+        if len(dirs) > NUM_TO_KEEP:
+            dirs_to_delete.extend(sorted(dirs, reverse=True)[NUM_TO_KEEP:])
+
+    if dirs_to_delete:
+        print 'Deleting the following directories:'
+        for d in dirs_to_delete:
+            d = os.path.join(mash_dir, d)
+            shutil.rmtree(d)
+            print d
```

Comment:
This code is duplicated with the script. We don't want to copy code as that violates the DRY Principle:

https://en.wikipedia.org/wiki/Don't_repeat_yourself

Instead, we want to import the existing code that does this and use it from the masher.